pkg/kv/dynamodb/store.go

```diff
@@ -238,6 +238,10 @@ func (s *Store) Get(ctx context.Context, partitionKey, key []byte) (*kv.ValueWit
 	})
 	const operation = "GetItem"
 	if err != nil {
+		if s.isSlowDownErr(err) {
+			s.logger.WithField("partition_key", partitionKey).WithContext(ctx).Error("get item: %w", kv.ErrSlowDown)
```

Comment:
The error is logged by the logging middleware. The log lines for returned errors don't add information and it could blow up easily.

Reply:
Same goes for all the below log lines :)

Reply:
The error isn't logged properly in the middleware since we don't return anywhere a `kv.ErrSlowDown`, therefore it's interpreted as an internal error and gets swallowed in it.
We don't want to return it as a `kv.ErrSlowDown` since we only want to monitor it and not change behavior.

Reply:
We have the prometheus metrics, what do we need the logging for?

Reply:
We can leave it as is, but we might want to investigate the behavior that led to the slowdown.
I guess that for testing that scenario we can see the flow of function calls that triggered it more clearly.
LMK if you still prefer that I would change it.

Reply:
I like logs. But a nit: it's a warning, not an error, as we didn't fail completely.